### PR TITLE
Fix some more analyzer tests

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit 571abfd41ea87fc9599709a95af660d0977644bd with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54128/571abfd41ea87fc9599709a95af660d0977644bd/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Planner/PlannerJoinTree.cpp [hunk 1]
```diff
@@ -208,6 +208,9 @@ bool applyTrivialCountIfPossible(
     if (row_policy_filter)
         return {};
 
+    if (select_query_info.additional_filter_ast)
```
Comment (novikd) [inline src/Planner/PlannerJoinTree.cpp:211]: What about additional result filters?

Comment (KochetovNicolai) [reply src/Planner/PlannerJoinTree.cpp:211]: The issue is that trivial count optimization ignores additional filer and creates a reading step with invalid count result.
Result filters does not have this issue: it is an expression on top of query result.

**Comment by KochetovNicolai** [comment]
The PR is so green that I have to merge it.
01576_alias_column_rewrite 01710_projection_additional_filters [are ok](https://s3.amazonaws.com/clickhouse-test-reports/54128/571abfd41ea87fc9599709a95af660d0977644bd/stateless_tests__release__analyzer_.html)

**Pull request merged**